WCAG 2.2 In brief - what - why (#196)
w3c/wai-intro-wcag: content/new-in-22.md

Provide ways to help users navigate, find content, and determine where they are.

### [AG version] 2.4.11 Focus Not Obscured (Minimum) (AA)
<section class="brief">
      <p>In brief:</p>
      <dl>
         <dt>Objective</dt><dd>Do not hide the item with focus</dd>
         <dt>Author task</dt><dd>The item with focus should be at least partially visible in the viewport</dd>
         <dt>Key beneficiaries</dt><dd>Some users with cognitive disabilities and sighted users reliant on keyboard interaction</dd>
      </dl>
  </section>

### [EOWG draft] 2.4.11 Focus Not Obscured (Minimum) (AA)
In brief:
<ul>

>   <li><p><span class="whatwhy">What to do: </span>Ensure that when items have focus, they are at least partially visible.</p> </li>

Reviewer: ```suggestion
  <li><p><span class="whatwhy">What to do: </span>Ensure the item receiving keyboard focus is at least partially visible.</p> </li>
```

Author: > Ensure the item getting keyboard focus is at least partially visible

It is shorter, though I think would be confusing to some -- "what item are you taking about?"

Remember EOWG proposes the In brief go **before** the SC wording in the Understanding docs &mdash; and that's how we have it in the [What's New doc](https://deploy-preview-196--wai-intro-wcag.netlify.app/standards-guidelines/wcag/new-in-22/#2411-focus-not-obscured-minimum-aa)

Reviewer: I can live with it either way. I'm not sure it's any less likely for someone to not have that question in the other phrasing, but can go with your preference

Author: OK. We could get other input if worth it? I'll resolve for now and we can re-open if warranted.

>   <li><p><span class="whatwhy">Why: </span>Some people cannot use a mouse and rely on seeing what has keyboard focus.</p> </li>

Reviewer: ```suggestion
  <li><p><span class="whatwhy">Why: </span>People who don't use a mouse need to see where they are on a page.</p> </li>
```

Reviewer: Decision point on "keyboard focus" versus "their location"

Reviewer: "where they're working"

Reviewer: ...to see where they're interacting with the page"

>   <li><p><span class="whatwhy">Why: </span>Some people cannot use a mouse and rely on seeing what has keyboard focus.</p> </li>

Author: From [Rebab](https://www.w3.org/2002/09/wbs/35532/wcag2_2_in_brief/results):
I see a benefit from adding the cognitive disabilities -``Key beneficiaries`` in the why. It resonates more: 
Key beneficiaries: Some users with cognitive disabilities and sighted users reliant on keyboard interaction.
Current why: Some people cannot use a mouse and rely on seeing what has keyboard focus.

Proposed why: Some users with cognitive disabilities and those who don`t use a mouse, rely on seeing the item that has keyboard focus on.

Reviewer: My concern with going this route is the same reason I think it made sense to remove the Key beneficiaries information. It becomes so simplified that it becomes almost non-representative. It's why I'm hoping for more detail in the Benefits area.

</blockquote>
<p><a href="https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum">Understanding Focus Not Obscured (Minimum)</a></p>

### [EOWG draft] 2.4.12 Focus Not Obscured (Enhanced) (AAA)

<p>In brief:</p>
<ul>

>       <li><p><span class="whatwhy">What to do: </span>Ensure that when items have focus, they are fully visible.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">What to do: </span>Ensure no part of the item receiving keyboard focus is hidden.</p></li>
```

Reviewer: FYI in WCAG it currently reads "Ensure no content obstructs the item receiving keyboard focus"
If there's any interest in putting the emphasis on blocking content, could always make this a bit more gracefully phrased as "Ensure content does not obstruct the item receiving keyboard focus" or something of that nature ("does not cover", "does not block", etc)

Author: Seems like it would be good to have similar wording for minimum and enhanced.

Reviewer: To clarify, the last two are variations on the same one, in case that wasn't clear. So the pair would be:
> Ensure the item getting keyboard focus is visible
 Ensure the item getting keyboard focus is fully visible

But I think I agree that the difference may not be enough

>       <li><p><span class="whatwhy">Why do it: </span>Some people cannot use a mouse and rely on seeing what has keyboard focus.</p> </li>

Reviewer: debated saying "clearly" or "fully". Elected to make same as minimum version

</ul>
<p><em>(Persona, problem, and works well same as 2.4.12 above.)</em></p>
<p class="sclabel">WCAG:</p>
<blockquote class="sc">
  <p>When a <a href="https://www.w3.org/TR/WCAG22/#dfn-user-interface-components">user interface component</a> receives keyboard focus, no part of the component is hidden by author-created content.</p>
</blockquote>
<p><a href="https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced">Understanding Focus Not Obscured (Enhanced)</a></p>

### [AG version] 2.4.13 Focus Appearance (AAA)
<section class="brief">
  <p>In brief:</p>
  <dl>
    <dt>Objective</dt>
    <dd>Make it easier to see what has keyboard focus</dd>
    <dt>Author task</dt>
    <dd>The focus indicator color contrasts 3:1 with both the component’s unfocused state and its surroundings</dd>
    <dt>Key beneficiaries</dt>
    <dd>Sighted users reliant on keyboard interaction</dd>
  </dl>
</section>

### [EOWG draft] 2.4.13 Focus Appearance (AAA)
<p>In brief:</p>
<ul>

>       <li><p><span class="whatwhy">What to do: </span>Ensure that the focus indicator contrasts 3:1 with the component’s unfocused state and with its surroundings.</p></li>

Author: From [Brian](https://www.w3.org/2002/09/wbs/35532/wcag2_2_in_brief/results):
I don't think we should have technical items in these descriptions, such as the 3:1 ratio. The description also seems to miss the need for the focus indication to be bolder (2px width). Can we have a less technical and more encompassing statement? 
"Ensure that the focus indicator is bold and has a high contrast with the component’s unfocused state and with its surroundings." or similar

Author: Well, we want it to be actionable, and this is the Understanding SC doc, so I think good to have technical detail where relevant and not confusing. I wonder if the idea of contrast ratios is well enough known among accessibility folks that it's OK?
I don't think of 2px width as "bold".
3:1 is not "high contrast" :-)

I added 2px to see what folks think...

>       <li><p><span class="whatwhy">What to do: </span>Ensure that the focus indicator contrasts 3:1 with the component’s unfocused state and with its surroundings.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">What to do: </span>Use a focus indicator of sufficient size and contrast.</p></li>
```

Author: agreed on Use a focus indicator of sufficient size and contrast.
@@SLH to update

>       <li><p><span class="whatwhy">Why users need it: </span>People with low contrast sensitivity cannot see small contrast differences.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">Why users need it: </span>People with low vision cannot see small or low-contrast indicators.</p></li>
```

Author: Consider including older users: "Older users and others with low contrast sensitivity cannot see small contrast differences."

Author: Some people can't see small contrast differences, including older people.

Author: discussed some versus many...

Many older people and others with low vision can't see small contrast differences.

>       <li><p><span class="whatwhy">Why users need it: </span>People with low contrast sensitivity cannot see small contrast differences.</p></li>

Reviewer: I'm finding the variability of the prefixes are making it harder to consistently assess these. For example, these first three have variations:
- why
- why do it
- why users need it

Obviously those lead to different wording (e.g., I'm not going to say "need" again if the prefix says "need").

I'm going to first incorporate the Why, since for the most part it is relatively straight forward to translate the currently adopted "Author task" to meet this (regardless of the prefix variation).

If EOWG can settle on their preference for the prefix, it'll help.
"What to do" seems to work fine to me (and is likely more meaningful to folks than 'author task')

There is undoubtedly a different timber to a prefix called "objective", or better, "goal", than one called "why". 
Having written those goals, I still would like to ask for the group's re-consideration on the value of a simple message like "Make it easier to spot the keyboard focus" or "Make controls easier to activate".

Author: > I'm finding the variability of the prefixes are making it harder to consistently assess these.

Sure. They are there so EOWG could see the different options. It's out for [survey](https://www.w3.org/2002/09/wbs/35532/wcag2_2_in_brief/?login) now -- the last question. That survey is open to anyone, so feel free to add your input. (We moved the closing date of that survey to Tuesday, which is a tighter timeline than EOWG is used to, so not sure how many replies we'll get.)

> I'm going to first incorporate the Why

I agree with some of your edit suggestions, though not all. Probably better that we resolve things here before you put it in WCAG space so we can discuss in one place.

Would you be willing to get on the phone to discuss these Tuesday?

Author: > Having written those goals, I still would like to ask for the group's re-consideration on the value of a simple message like "Make it easier to spot the keyboard focus" or "Make controls easier to activate".

Do you mean in addition to the what and why? So:
* Goal:
* What to do:
* Why:

or instead of Why?

Depending on which, I'll have additional input. :-)

<hr>

Though I will say now that "easier" seems not to convey the importance for some users.

Reviewer: Suggesting instead of. I'm just musing.
I'm also exploring the mashup of the goal with the target users, as suggested somewhere in the conversation.
I'm going to let my mind percolate on this over part of the weekend.

Reviewer: > I agree with some of your edit suggestions, though not all. Probably better that we resolve things here before you put it in WCAG space so we can discuss in one place.

@shawna-slh I have not made any changes in the WCAG In Briefs based on our iterations in this file; however, some of the Author task phrases were updated in WCAG already based on user feedback, so some of the changes I've made to 'what to do' are based on that existing WCAG wording.

Author: Based on EOWG input -- and the fact that many people reading the Understanding docs don't understand the needs of people with disabilities -- going with "Why it's important" for now... to see how that fits.

Make it easier for users to operate functionality through various inputs beyond keyboard.

### [AG version] 2.5.7 Dragging Movements (AA)

<section class="brief">
<p>In brief:</p>
<dl>
         <dt>Objective</dt><dd>Don't rely on dragging for user actions</dd>
         <dt>Author task</dt><dd>Provide a simple pointer alternative to any action that involves dragging</dd>
         <dt>Key beneficiaries</dt><dd>Users with some physical disabilities</dd>
      </dl>
 </section>

### [EOWG draft] 2.5.7 Dragging Movements (AA)
<p>In brief:</p>
   <ul>

>       <li><p><span class="whatwhy">What to do: </span>For any action that involves dragging, provide a simple pointer alternative.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">What to do: </span>Provide a simple pointer alternative to any action that involves dragging.</p></li>
```

Author: Mike and Shawn discussed. Agreed to leave as is.

Make Web pages appear and operate in predictable ways.

### [AG version] 3.2.6 Consistent Help (A)

<section class="brief">
<p>In brief:</p>
   <dl>
          <dt>Objective</dt><dd>Consistently locate user help</dd>
          <dt>Author task</dt><dd>If you provide a help mechanism, provide it in a consistent location</dd>
          <dt>Key beneficiaries</dt><dd>Users with some cognitive disabilities</dd>
      </dl>
</section>

### [EOWG draft] 3.2.6 Consistent Help (A)

<p>In brief:</p>
   <ul>

>       <li><p><span class="whatwhy">What to do: </span>If you provide help, put  it in the same place on all web pages and app windows.</p></li>

Reviewer: Still fiddling around with this one...
```suggestion
      <li><p><span class="whatwhy">What to do: </span>Consistently locate any user help that appears on multiple pages.</p></li>
```

Author: From [Kris Anne](https://www.w3.org/2002/09/wbs/35532/wcag2_2_in_brief/results#xq8):
... People who need help are more likely to find it if it is in a consistent place.

Reviewer: Put help in the same place when it appears on multiple pages

>       <li><p><span class="whatwhy">Why: </span>People who have difficulty finding help are more likely to find it when it is in the same place.</p></li>

Author: went with: People who need help can find it more easily if it's in the same place.

<p><a href="https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry">Understanding Redundant Entry</a></p>

### 3.3.8 Accessible Authentication (Minimum) (AA)
<p>In brief:</p>
   <ul>
      <li><p><span class="whatwhy">What to do: </span>Don’t make people memorize or transcribe something to log in.</p></li>

>       <li><p><span class="whatwhy">Why it's important: </span>Some people with cognitive disabilities cannot memorize a username and password, or type them.</p></li>

Reviewer: Since we say "memorize" under the what to do, does it add value saying "remember" here instead? I believe it is synonymously close enough that it adds value

```suggestion
      <li><p><span class="whatwhy">Why it's important: </span>Some people with cognitive disabilities cannot remember a username and password, or type them.</p></li>
```

Reviewer: Or how about "recall"?

Help users avoid and correct mistakes.

### [AG version] 3.3.7 Redundant Entry (A)

> _[@@ SLH didn't find In Brief for this one.]_

Reviewer: Looks like it was lost in some change!

### [EOWG draft] 3.3.7 Redundant Entry (A)

<p>In brief:</p>
   <ul>

>       <li><p><span class="whatwhy">What to do: </span>If you have data that needs to be entered again, auto-populate it or make it avaliable for the user to select.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">What to do: </span>Don't ask the same information twice in the same process.</p></li>
```

</blockquote>
<p><a href="https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry">Understanding Redundant Entry</a></p>

### [AG version] 3.3.8 Accessible Authentication (Minimum) (AA)

<section class="brief">
  <p>In brief:</p>
  <dl>
    <dt>Objective</dt>
    <dd>Make logins possible with less mental effort</dd>
    <dt>Author task</dt>
    <dd>Don’t make people memorize or transcribe something in order to log in</dd>
    <dt>Key beneficiaries</dt>
    <dd>Users with some cognitive disabilities</dd>
  </dl>
</section>

### [EOWG draft] 3.3.8 Accessible Authentication (Minimum) (AA)
<p>In brief:</p>
   <ul>

>       <li><p><span class="whatwhy">What to do: </span>Don’t make people memorize or transcribe something in order to log in.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">What to do: </span>Don’t make people memorize or transcribe something to log in.</p></li>
```

</dl>
  <p class="note">Note: "Object recognition" and "Personal content" may be represented by images, video, or audio.</p>
  <p class="note">Note: Examples of mechanisms that satisfy this criterion include:</p>
   <ol>
      <li>support for password entry by password managers to reduce memory need, and</li>
      <li>copy and paste to reduce the cognitive burden of re-typing.</li>
    </ol>
</blockquote>
<p><a href="https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication">Understanding Accessible Authentication (Minimum)</a></p>

### [AG version] 3.3.9 Accessible Authentication (Enhanced) (AAA)

<section class="brief">
<p>In brief</p>
   <dl>
            <dt>Objective</dt><dd>Make logins possible with less mental effort</dd>
            <dt>Author task</dt><dd>Don’t make people recognize objects or user-supplied information in order to login</dd>
            <dt>Key beneficiaries</dt><dd>Users with some cognitive disabilities</dd>
        </dl>
 </section>

### [EOWG draft] 3.3.9 Accessible Authentication (Enhanced) (AAA)

<p>In brief:</p>
   <ul>

>       <li><p><span class="whatwhy">What to do: </span>Don’t make people recognize objects or information they previous provided in order to login.</p></li>

Reviewer: ```suggestion
      <li><p><span class="whatwhy">What to do: </span>Don’t make people recognize objects or information in order to login.</p></li>
```
